manifests/init.pp

```diff
@@ -156,19 +156,7 @@
     Anchor['php::end']
   }
 
-  if $fpm {
-    Anchor['php::begin'] ->
-      class { '::php::fpm':
-        service_enable   => $fpm_service_enable,
-        service_ensure   => $fpm_service_ensure,
-        service_name     => $fpm_service_name,
-        service_provider => $fpm_service_provider,
-        settings         => $real_settings,
-        log_owner        => $log_owner,
-        log_group        => $log_group,
-      } ->
-    Anchor['php::end']
-  }
+  if $fpm { contain '::php::fpm' }
```

Comment:
All these parameters will now just default correctly to the values from the base class.  You can now override them in hiera if you really want.  We're probably alright switching to `contain` now (puppet >= 3.4)?